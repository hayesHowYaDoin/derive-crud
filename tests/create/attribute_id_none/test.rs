use derive_crud::Create;

#[derive(Create)]
#[crud_table("test_table")]
struct Test {
    id: i64,
    name: String,
}

#[tokio::main]
async fn main() {
    Test::create("Debbie".to_string()).await;
}
