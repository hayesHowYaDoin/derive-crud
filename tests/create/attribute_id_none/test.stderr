error: A struct field must be annotated with `#[crud_id]` to specify the database table
 --> tests/create/attribute_id_none/test.rs:5:13
  |
5 |   struct Test {
  |  _____________^
6 | |     id: i64,
7 | |     name: String,
8 | | }
  | |_^

error[E0599]: no function or associated item named `create` found for struct `Test` in the current scope
  --> tests/create/attribute_id_none/test.rs:12:11
   |
5  | struct Test {
   | ----------- function or associated item `create` not found for this struct
...
12 |     Test::create("Debbie".to_string()).await.expect("Failed to create entry");
   |           ^^^^^^ function or associated item not found in `Test`
